scripts/ci-tool.py

```diff
@@ -327,9 +325,30 @@ def main():
     }
 
     if args.build or args.lint or args.all:
-        cmake_configure(project_root, build_root, options)
-
-    if args.build or args.all:
+        # choose the generater initially based on what already exists there
+        if isdir(build_root):
+            if isfile(join(build_root, 'Makefile')):
+                generator = 'Unix Makefiles'
+            elif isfile(join(build_root, 'build.ninja')):
+                generator = 'Ninja'
+            else:
+                raise RuntimeError('Unable to detect existing generator type')
+        else:
+
+            # in the case of a new build prefer Ninja over make (because it is faster)
```

Comment:
👍 - this is kindof transparent to the user anyway right?

Reply:
Correct